scripts/enable-ecs-agent-gpu-support.sh

```diff
@@ -22,8 +23,75 @@ enabled=1
 exclude=libglvnd-*
 EOF
 
-# this repo is temporary and only used for installing the system-release-nvidia package
+# the amzn2-nvidia repo is temporary and only used for installing the system-release-nvidia package
 sudo mv $tmpfile /etc/yum.repos.d/amzn2-nvidia-tmp.repo
+
+# only install open driver for post-kepler gpus
+if [[ $AMI_TYPE != "al2keplergpu" ]]; then
+    sudo yum install -y yum-plugin-versionlock \
+        yum-utils
+    sudo amazon-linux-extras install epel -y
+    sudo yum install -y "kernel-devel-uname-r == $(uname -r)"
+
+    # pull nvidia version from what's available in amzn2-nvidia
+    # trim after `:` until `-` to get the major.minor.patch version
+    NVIDIA_VERSION=$(yum list available | grep nvidia-kmod-common | awk '{print $2}' | sed -e 's/.*://' -e 's/-.*//')
+
+    # disable amzn2 in favor of rh repo
+    sudo yum-config-manager --disable amzn2-nvidia
+    sudo yum-config-manager --add-repo=https://developer.download.nvidia.com/compute/cuda/repos/rhel7/x86_64/cuda-rhel7.repo
+    sudo yum-config-manager --enable cuda-rhel7.repo
+
+    # install open dkms from rh repo
+    sudo yum install -y nvidia-kmod-common-${NVIDIA_VERSION}
+
+    # build nvidia-open kmod tar
+    DKMS=/usr/sbin/dkms
+    DKMS_ARCHIVE_DIR=/var/lib/dkms-archive
+    MODULE_NAME="nvidia-open"
+    MODULE_VERSION=$(${DKMS} status -m ${MODULE_NAME} | awk '{print $2}' | tr -d ',:')
+
+    sudo ${DKMS} build -m "${MODULE_NAME}" -v "${MODULE_VERSION}"
+    sudo ${DKMS} mktarball -m "${MODULE_NAME}" -v "${MODULE_VERSION}"
+    sudo mkdir -p "${DKMS_ARCHIVE_DIR}/${MODULE_NAME}/"
+    sudo cp /var/lib/dkms/${MODULE_NAME}/${MODULE_VERSION}/tarball/*.tar.gz "${DKMS_ARCHIVE_DIR}/${MODULE_NAME}/"
+
+    # re-enable amzn2 and clean up
+    sudo yum remove -y kmod-nvidia-open-dkms
+    sudo yum-config-manager --disable cuda-rhel7.repo
+    sudo rm /etc/yum.repos.d/cuda-rhel7.repo
+    sudo rm -rf /var/cache/yum
+    sudo yum-config-manager --enable amzn2-nvidia
+
+    # copy install-nvidia-open-kmod.sh to host
+    sudo mkdir -p /var/lib/ecs/scripts
+
+    tmpfile=$(mktemp)
+    cat >$tmpfile <<"EOF"
+#!/usr/bin/env bash
```

Comment:
Do we need to test if this script is idempotent (works for instance restart)?

Reply:
we don't -- the script should only be run once as part of userdata at instance startup.